core/src/main/java/com/segment/analytics/kotlin/core/utilities/JSON.kt

```diff
@@ -181,3 +181,27 @@ fun JsonObject.transformKeys(transform: (String) -> String): JsonObject {
 fun JsonObject.transformValues(transform: (JsonElement) -> JsonElement): JsonObject {
     return JsonObject(this.mapValues { transform(it.value) })
 }
+
+// Utility function to update a JsonObject. The updated copy is returned. Original copy is not touched
+fun updateJsonObject(jsonObject: JsonObject, closure: (MutableMap<String, JsonElement>) -> Unit): JsonObject {
+    val content = jsonObject.toMutableMap()
+    closure(content)
+    return JsonObject(content)
+}
+
+operator fun MutableMap<String, JsonElement>.set(key:String, value: String?) {
```

Comment:
would adding these APIs on such a widely used type (like MutableMap) have consequences?

Reply:
yeah. that's a valid concern. the good thing is that it only overrides set operator for `MutableMap<String, JsonElement>`, not the other types. I tried `MutableMap<String, String>`, it's not impacted. so the impact is limited.